src/build/app/mod.rs

```diff
@@ -1019,7 +1019,6 @@ impl<'help> App<'help> {
     ///     .get_matches();
     /// ```
     /// [`ColorChoice::Auto`]: crate::ColorChoice::Auto
-    #[cfg(feature = "color")]
```

Comment:
Why is this removed?

Reply:
I wish github made commit messages more visible

> fix: Allow no-op API for color 
>
> This is mostly helpful for tests since they want to force no color in
some cases and be independent of the `color` feature.  We could add some
boiler plate to help but that is a lot harder in `clap_derive`.

In general, conditional APIs are a pain to deal with.  If there is reasonable behavior for an API to be be a no-op, I think we should prefer offering it.

Reply:
Sorry, I didn't realise it had more description in the commit message.

> This is mostly helpful for tests since they want to force no color insome cases and be independent of the `color` feature. 

I am not really sure that I understood that. Can you please expand more?

> In general, conditional APIs are a pain to deal with.

While I do agree, I think the benefit of gating this behind a feature is actually more to the end-user who's reading the docs and does not get confused about why this API is there in the default code when there's a specific `color` feature available.

But now that I am saying this, I see that we always offer `ColorChoice` to be public which might also confuse the users.

Reply:
> > This is mostly helpful for tests since they want to force no color insome cases and be independent of the color feature.

> I am not really sure that I understood that. Can you please expand more?

Let's take a recent test I added:
```rust
#[test]
fn derive_generated_error_has_full_context() {
    #[derive(Debug, Parser)]
    #[clap(setting(AppSettings::SubcommandsNegateReqs), color = ColorChoice::Never)]
    struct Opts {
        #[clap(long)]
        req_str: String,

        #[clap(subcommand)]
        cmd: Option<SubCommands>,
    }

    #[derive(Debug, Parser)]
    enum SubCommands {
        Sub {
            #[clap(short, long, parse(from_occurrences))]
            verbose: u8,
        },
    }

    let result = Opts::try_parse_from(&["test", "sub"]);
    assert!(
        result.is_err(),
        "`SubcommandsNegateReqs` with non-optional `req_str` should fail: {:?}",
        result.unwrap()
    );

    let expected = r#"error: The following required argument was not provided: req-str

USAGE:
    clap_derive --req-str <REQ_STR>
    clap_derive <SUBCOMMAND>

For more information try --help
"#;
    assert_eq!(result.unwrap_err().to_string(), expected);
}
```
Without `color = ColorChoice::Never`, running `Error::to_string`  might  include colored formatting, which makes testing the result difficult.  In this case, I want color disabled, independent of the `color` feature flag.

There are probably alternative ways of handling this but the ones I tried didn't quite work because of the circular dependency nature between `clap_derive` and `clap`.  Ideally, we could detect what features are enabled on `clap` and respond to those.

Reply:
> Without `color = ColorChoice::Never`, running Error::to_string might include colored formatting, which makes testing the result difficult. In this case, I want color disabled, independent of the color feature flag.

I just want to confirm that color wasn't automatically turned off since it couldn't detect a tty? I am pretty sure color in testing was never a problem.

Reply:
Ok, dug into this a bit.  Its not about tty detection (because `--nocapture` would then fail which it doesn't) but the fact that we are wrapping `Colorizer::fmt` which never reports color.  We only support colorizing when writing directly to stdout/stderr and never to an arbitrary target.  I've instead changed this to remove color disabling from two tests.

Reply:
(I still think we should be cautious on feature gating APIs, especially types, but that is now a separate conversation from this PR)